packages/experiment-core/src/transport/stream.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type StreamErrorEvent = { message?: string; status?: number };
+export type StreamOpenEvent = { type?: string };
+export type StreamMessageEvent = { data?: string };
+export type StreamEvent =
+  | StreamErrorEvent
+  | StreamOpenEvent
+  | StreamMessageEvent;
+
+/**
+ * The EventSource client interface.
+ * https://html.spec.whatwg.org/multipage/server-sent-events.html#eventsource
+ */
+export interface StreamEventSource {
+  readonly url: string;
+  readonly readyState: number;
+  readonly withCredentials: boolean;
+  onopen: (evt: StreamOpenEvent) => any;
+  onmessage: (evt: StreamMessageEvent) => any;
+  onerror: (evt: StreamErrorEvent) => any;
+  // The following are in specs but not currently used. Uncomment as we start to use them.
+  //   addEventListener(type: string, listener: (evt: StreamEvent) => void): void;
+  //   dispatchEvent(evt: Event): boolean;
+  //   removeEventListener(type: string, listener: (evt: StreamEvent) => void): void;
+  close(): void;
+}
+
+// A type that can new.
+// Ex. import EventSource from 'eventsource'; let a = EventSource; let b = new a("", {});
+type Class<T> = new (...args: any[]) => T;
+/**
+ * This is the type for the class of StreamEventSource.
+ * It should support new operation along with defined static members.
+ */
+export type StreamEventSourceClass = Class<StreamEventSource> & {
+  // These are static members usable without new.
+  readonly CLOSED: number;
+  readonly CONNECTING: number;
+  readonly OPEN: number;
+};
```

Comment:
Interface with 3 fields requires instantiating a new object to access them, similar to instance variables. having this on type is similar to static variables. 
Was trying to limit access to those on the type instead of objects. Since we may use a factory function to create them, they can just access them using the objects.